clang/lib/Frontend/CompilerInvocation.cpp

```diff
@@ -3381,6 +3385,19 @@ static bool ParseFrontendArgs(FrontendOptions &Opts, ArgList &Args,
     Opts.ModuleCacheKeys.emplace_back(Values[0], Values[1]);
   }
 
+  // Handle path prefix mappings
+  for (const Arg *A : Args.filtered(OPT_fdepscan_prefix_map)) {
+    ArrayRef<const char *> Values = A->getValues();
+    assert(Values.size() == 2);
+    Opts.PathPrefixMappings.emplace_back(Values[0], Values[1]);
+  }
+  for (const Arg *A : Args.filtered(OPT_fdepscan_prefix_map_EQ)) {
+    StringRef Val = A->getValue();
+    if (std::optional<llvm::MappedPrefix> mapping = llvm::MappedPrefix::getFromJoined(Val)) {
```

Comment:
This should probably have a diagnostic if it fails. I guess this was already a problem before this patch though?  Also moot if we make it driver-only.